frontend/components/Sider.js

```diff
@@ -1,19 +1,56 @@
+import { useState, useEffect } from 'react';
+import Image from 'next/image';
 import { Layout, Menu } from 'antd';
+
 import { MENU_ITEMS } from '../config/Sider.config';
 
 const { Sider: AntdSider } = Layout;
 
-const Sider = ({ collapsed, userType }) => (
-  <AntdSider trigger={null} collapsible collapsed={collapsed}>
-    <div className='h-8 bg-gray-400 bg-opacity-30 m-4 mb-8' />
-    <Menu theme="dark" mode="inline" defaultSelectedKeys={['0']}>
-      {MENU_ITEMS[userType].map(({ text, icon }, index) => (
-        <Menu.Item key={index} icon={icon}>
-          {text}
-        </Menu.Item>
-      ))}
-    </Menu>
-  </AntdSider>
-);
+const Sider = ({ collapsed, userType }) => {
+  const [isBroken, setIsBroken] = useState(collapsed);
+  const [showLogoText, setShowLogoText] = useState(!collapsed);
+
+  useEffect(() => {
+    if (collapsed) {
+      setTimeout(() => setShowLogoText(!collapsed), 30)
+    } else {
+      setTimeout(() => setShowLogoText(!collapsed), 145)
+    }
+  }, [collapsed])
```

Comment:
smoothing responsive logo on collapsing

Reply:
In another way, we can use `collapsed` to toggle className like `hidden` and add a transition to it.

Reply:
In this case, when collapse changes true->false and false->true, they use different delay time and need highly customized delay time. Transition delay might not be able to deal with it.

Reply:
That's ok.